samples/inputs/mask-input/value-modes/src/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import { IgrMaskInput, IgrIcon, IgrRadioGroup, IgrRadio, IgrMaskInputModule, IgrIconModule, IgrRadioGroupModule, IgrRadioModule } from 'igniteui-react';
+import 'igniteui-webcomponents/themes/light/bootstrap.css';
+
+IgrMaskInputModule.register();
+IgrIconModule.register();
+IgrRadioGroupModule.register();
+IgrRadioModule.register();
+
+export default class MaskInputValueModes extends React.Component<any, any> {
+
+    public maskRef: IgrMaskInput;
+    public fileIcon: IgrIcon;
+
+    constructor(props: any) {
+        super(props);
+        this.onMaskRef = this.onMaskRef.bind(this);
+        this.iconFileRef = this.iconFileRef.bind(this);
+        this.state = {value: ""};   
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="container sample center">
+                <IgrMaskInput ref={this.onMaskRef}>
```

Comment:
You can add a property inputOccured here and update this.state.value dynamically.